src/analysis/processing/qgsalgorithmmultiintersection.cpp

```diff
@@ -0,0 +1,214 @@
+  addParameter( prefix.release() );
+
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Intersection" ) ) );
+}
+
+QVariantMap QgsMultiIntersectionAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  std::unique_ptr< QgsFeatureSource > sourceA( parameterAsSource( parameters, QStringLiteral( "INPUT" ), context ) );
+  if ( !sourceA )
+    throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "INPUT" ) ) );
+
+  const QList< QgsMapLayer * > layers = parameterAsLayerList( parameters, QStringLiteral( "OVERLAYS" ), context );
+
+  // loop through overlay layers and check whether they are vectors
+  long totalLayerCount = 0;
+  for ( QgsMapLayer *layer : layers )
+  {
+    if ( feedback->isCanceled() )
+      break;
+
+    if ( !layer )
+      throw QgsProcessingException( QObject::tr( "Error retrieving map layer." ) );
+
+    if ( layer->type() != QgsMapLayerType::VectorLayer )
+      throw QgsProcessingException( QObject::tr( "All layers must be vector layers!" ) );
+
+    totalLayerCount++;
+  }
+
+  const QString overlayFieldsPrefix = parameterAsString( parameters, QStringLiteral( "OVERLAY_FIELDS_PREFIX" ), context );
+
+  const QgsWkbTypes::Type geometryType = QgsWkbTypes::multiType( sourceA->wkbType() );
+  const QgsCoordinateReferenceSystem crs = sourceA->sourceCrs();
+  std::unique_ptr< QgsFeatureSink > sink;
+  long count = 0;
+  QVariantMap outputs;
+
+  QList<int> fieldIndicesA, fieldIndicesB;
+  QgsFields outputFields;
+
+  if ( totalLayerCount == 1 )
+  {
+    QgsVectorLayer *overlayLayer = qobject_cast< QgsVectorLayer * >( layers.at( 0 ) );
+
+    fieldIndicesA = QgsProcessingUtils::fieldNamesToIndices( QStringList(), sourceA->fields() );
+    fieldIndicesB = QgsProcessingUtils::fieldNamesToIndices( QStringList(), overlayLayer->fields() );
+
+    outputFields = QgsProcessingUtils::combineFields(
+                     QgsProcessingUtils::indicesToFields( fieldIndicesA, sourceA->fields() ),
+                     QgsProcessingUtils::indicesToFields( fieldIndicesB, overlayLayer->fields() ),
+                     overlayFieldsPrefix );
+
+    QString dest;
+    sink.reset( parameterAsSink( parameters, QStringLiteral( "OUTPUT" ), context, dest, outputFields, geometryType, crs, QgsFeatureSink::RegeneratePrimaryKey ) );
+    if ( !sink )
+      throw QgsProcessingException( invalidSinkError( parameters, QStringLiteral( "OUTPUT" ) ) );
+
+    outputs.insert( QStringLiteral( "OUTPUT" ), dest );
+
+    const long total = sourceA->featureCount();
+    QgsOverlayUtils::intersection( *sourceA, *overlayLayer, *sink, context, feedback, count, total, fieldIndicesA, fieldIndicesB );
+  }
+  else
+  {
+    QgsProcessingMultiStepFeedback multiStepFeedback( totalLayerCount, feedback );
+
+    QString id = QStringLiteral( "memory:" );
```

Comment:
same note as previous -- it'd be easier to quickly read the code if this was just hardcoded in lines 167/201

Reply:
Here the same situation, having id in a variable is necessary as later we will use it to retrieve layer.